src/main/java/ui/components/FilterTextField.java

```diff
@@ -87,22 +88,39 @@ private void setup() {
                     cancel.run();
                 } else {
                     revertEdit();
-                    selectAll();
                 }
             }
         });
     }
 
-    private void performCompletion(KeyEvent e) {
-        String word = getCurrentWord() + e.getCharacter();
+    /**
+     * Completion is only started when there's effectively nothing (only whitespace)
+     * after the caret, or if there is selected text (indicating either that we are in
+     * the midst of completion, or that the user does not want the selected text and is
+     * typing to replace it)
+     * @return true if completion should be started
+     */
+    private boolean shouldStartCompletion() {
+        return getCharAfterCaret().trim().isEmpty() || !getSelectedText().isEmpty();
```

Comment:
I need confirmation on this behaviour. What if the user select text in the middle of the filter? Should completion be performed? Say `assignee:me milestone:current` and the selected text is `ignee` of `assignee`.